### PR TITLE
Add more logging

### PR DESCRIPTION
**Pull request by anuragsoni**

### CODE REVIEW

File: blue_http.opam [hunk 1]
```diff
@@ -11,6 +11,7 @@ homepage: "https://github.com/brendanlong/blue-http"
 doc: "https://brendanlong.github.io/blue-http"
 bug-reports: "https://github.com/brendanlong/blue-http/issues"
 depends: [
+  "core"
```
Comment (brendanlong) [inline blue_http.opam]: Can you sort this below `bisect_ppx`?

Also doesn't cohttp-async already depend on core?

File: dune-project [hunk 2]
```diff
@@ -21,6 +21,7 @@
  (description
    "Blue_http builds on Cohttp but provides TLS support with certification validation, connection pooling, and redirect support out of the box.")
  (depends
+   core
```
Comment (brendanlong) [inline dune-project]: This should also be sorted and formatted

File: src/logger.ml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+let key = Univ_map.Key.create ~name:"logging_uuid" Uuid.sexp_of_t
+let with_uuid uuid f = Scheduler.with_local key (Some uuid) ~f
+
+let with_uuid_tags tags =
+  let open Option.Let_syntax in
+  let tag =
+    let%map uuid = Scheduler.find_local key in
+    "request_uuid", Uuid.to_string uuid
+  in
+  Option.(to_list tag @ value ~default:[] tags)
```
Comment (brendanlong) [inline src/logger.ml]:
```suggestion
  match Scheduler.find_local key with
  | None -> tags
  | Some uuid ->
    ("request_uuid", Uuid.to_string uuid) :: (Option.value ~default:[] tags)
```

File: blue_http.opam (hunk 1 above)
Comment (anuragsoni) [reply blue_http.opam]: I think it does. I've removed this entry

**Review by brendanlong — APPROVED** [review]

**Pull request merged**